src/usr/share/netcheck/netcheck.py

```diff
@@ -74,7 +72,8 @@ def _dns_query(self, network, nameserver, query_name):
         success = False
 
         # TODO: Add support for when this is not able to obtain the IP.
-        interface_ip = self.network_meta.get_interface_ip(network)
+        # TODO: Try to use the gateway address instead of the interface address.
```

Comment:
As I said, I can't fully remember why I asked you to add this comment. The only thing I can think of is that maybe you can avoid having the NetworkManagerHelper.network_id_table dictionary somehow if you can use the gateway address instead of the interface address.

If I don't come up with a better explanation before I am finished with this review (and the above doesn't make sense), please remove this TODO.

Reply:
OK, I figured this out. In NetworkManagerHelper.get_network_ip, you are getting the first IP address, but that might not be the IP that is associated with the gateway. In that event, the DNS lookup below will fail. You should really use either the gateway address (if that works) or the IP address that is associated with the gateway to do the DNS lookup.